src/main/java/past/cwdt/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -30,9 +35,37 @@ public BookDto getBookById(Long id) {
     }
 
     @Override
-    public List<BookDto> findAll() {
-        return bookRepository.findAll().stream()
+    public List<BookDto> findAll(Pageable pageable) {
+        return bookRepository.findAll(pageable).stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        checkIfBookExistsById(id);
+        Book book = bookMapper.toBook(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
this is wrong

you should first get the book by id

then update it via mapstruct:
```
void updateBook(CreateBookRequestDto dto, @MappingTarget Book book);
```
and then save the updated book